build/lex.go

```diff
@@ -27,50 +27,36 @@ import (
 )
 
 // BuildFilenames is a collection of filenames in lowercase that are treated as BUILD files.
-var BuildFilenames = map[string]bool{
-	"build": true,
-	"build.bazel": true,
-	"workspace": true,
-	"workspace.bazel": true,
-	"stdin": true,
+var BuildFilenames = map[string]FileType{
+	"buck":            BUCK,
+	"build":           BUILD,
+	"build.bazel":     BUILD,
+	"workspace":       BUILD,
+	"workspace.bazel": BUILD,
+	"stdin":           BUILD,
 }
 
-// ParseBuild parses a file, marks it as a BUILD file and returns the corresponding parse tree.
-//
-// The filename is used only for generating error messages.
-func ParseBuild(filename string, data []byte) (*File, error) {
-	in := newInput(filename, data)
-	f, err := in.parse()
-	if f != nil {
-		f.Build = true
+// Parse parses the input data and returns the corresponding parse tree.
+func Parse(filename string, fileType FileType, data []byte) (*File, error) {
```

Comment:
Could you please not change the signature (at least for now)? There are existing projects that use buildifier as a library, and they have calls to `build.Parse(filename string, data []byte)`, such change will break all of them at once.

If you want to change the signature, consider adding alternative API (for example, a function called `ParseFile` with a new set of parameters) while keeping the existing functions for compatibility.